qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -0,0 +1,394 @@
+                probability_output_dict["1"] = counts.get("1", 0) / sum(counts.values())
+                probability_output_dict["0"] = 1 - probability_output_dict["1"]
+            output_dict_list.append(probability_output_dict)
+
+        return output_dict_list
+
+
+class MockIQFineAmpHelper(MockIQExperimentHelper):
+    """Functions needed for Fine Amplitude experiment on mock IQ backend"""
+
+    def __init__(self, angle_error: float = 0, angle_per_gate: float = 0, gate_name: str = "x"):
+        self.angle_error = angle_error
+        self.angle_per_gate = angle_per_gate
+        self.gate_name = gate_name
+
+    def compute_probabilities(self, circuits: List[QuantumCircuit]) -> List[Dict[str, float]]:
+        """Return the probability of being in the excited state."""
+        angle_error = self.angle_error
+        angle_per_gate = self.angle_per_gate
+        gate_name = self.gate_name
+        output_dict_list = []
+        for circuit in circuits:
+            probability_output_dict = {}
+            n_ops = circuit.count_ops().get(gate_name, 0)
+            angle = n_ops * (angle_per_gate + angle_error)
+
+            if gate_name != "sx":
+                angle += np.pi / 2 * circuit.count_ops().get("sx", 0)
+
+            if gate_name != "x":
+                angle += np.pi * circuit.count_ops().get("x", 0)
+
+            # Dictionary of output string vectors and their probability
+            probability_output_dict["1"] = np.sin(angle / 2) ** 2
+            probability_output_dict["0"] = 1 - probability_output_dict["1"]
+            output_dict_list.append(probability_output_dict)
+
+        return output_dict_list
+
+
+class MockIQRamseyXYHelper(MockIQExperimentHelper):
+    """Functions needed for Ramsey XY experiment on mock IQ backend"""
+
+    def __init__(self, t2ramsey: float = 100e-6, freq_shift: float = 0):
+        self.t2ramsey = t2ramsey
+        self.freq_shift = freq_shift
+
+    def compute_probabilities(self, circuits: List[QuantumCircuit]) -> List[Dict[str, float]]:
+        """Return the probability of being in the excited state."""
+        t2ramsey = self.t2ramsey
+        freq_shift = self.freq_shift
+        output_dict_list = []
+        for circuit in circuits:
+            probability_output_dict = {}
+            series = circuit.metadata["series"]
+            delay = circuit.metadata["xval"]
+
+            if series == "X":
+                phase_offset = 0.0
+            else:
+                phase_offset = np.pi / 2
+
+            probability_output_dict["1"] = (
+                0.5
+                * np.exp(-delay / t2ramsey)
+                * np.cos(2 * np.pi * delay * freq_shift - phase_offset)
+                + 0.5
+            )
+            probability_output_dict["0"] = 1 - probability_output_dict["1"]
+            output_dict_list.append(probability_output_dict)
+        return output_dict_list
+
+
+class MockIQResonatorSpectroscopyHelper(MockIQExperimentHelper):
```

Comment:
Would it make sense to make this inherit from the spectroscopy helper to avoid some code duplication? This would require adding a variable to tell the class which gate name to look for.

Reply:
Yea, I deleted the class `MockIQResonatorSpectroscopyHelper` and I added `gate_name` attribute to `MockIQSpectroscopyHelper` class.
In the probability calculation, I added the following if:
```python
  if self.gate_name == "measure":
      freq_shift = (
          next(iter(circuit.calibrations[self.gate_name].values())).blocks[0].frequency
      )
  elif self.gate_name == "Spec":
      freq_shift = next(iter(circuit.calibrations[self.gate_name]))[1][0]
  else:
      raise ValueError("The gate name %s isn't supported.", self.gate_name)
```